### PR TITLE
sxhkd: configurable package and command line arguments

### LINKED ISSUE #1598: Sxhkd module can't handle non-QWERTY keyboard layouts
### Issue description

The [sxhkd manual](https://github.com/baskerville/sxhkd/blob/master/doc/sxhkd.1.asciidoc) says that if you're using a non-QWERTY keyboard layout, you should run sxhkd with the `-m 1` option, so that it doesn't ignore keymap signals. But the command with which sxhkd is run is not configurable with this module. 

The solution, I'm guessing, would be to allow the service to be run with extra flags. Or, more semantically, to add an option that supports non-QWERTY layouts. 

#### Maintainer CC

@nbardiuk

#### Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.2, NixOS, 21.03pre250878.a52e974cff8 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(root): `"nixos-21.03pre250878.a52e974cff8"`
 - channels(jon): `"home-manager, nixpkgs-21.03pre250049.4b30d5601da"`
 - nixpkgs: `/home/jon/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by berbiche**
### Description

Fixes #1598.

@JonathanReeve I don't use sxhkd, can you confirm this PR resolves your issue?

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/sxhkd.nix [hunk 1]
```diff
@@ -22,6 +22,21 @@ in
...
     enable = mkEnableOption "simple X hotkey daemon";
 
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sxhkd;
+      defaultText = "pkgs.sxhkd";
+      description = "Package containing the <command>sxhkd</command> executable.";
+    };
+
+    commandline = mkOption {
```
Comment (rycee) [inline modules/services/sxhkd.nix:32]: We have elsewhere called this `extraOptions`, e.g.
https://github.com/nix-community/home-manager/blob/7e5fee4268f53be8758598b7634dff8b3ad8a22b/modules/services/redshift.nix#L112-L120

So perhaps good to reuse that name here?

File: modules/services/sxhkd.nix [hunk 2]
```diff
@@ -22,6 +22,21 @@ in
...
+      type = types.either types.str (types.listOf types.str);
```
Comment (rycee) [inline modules/services/sxhkd.nix:33]: I would suggest simply having `listOf str` here and avoid the automatic conversion.

Comment (berbiche) [reply modules/services/sxhkd.nix]: Fixed

**Comment by JonathanReeve** [comment]
I don't have a test setup at the moment, but the code looks good to me!

**Comment by rycee** [comment]
Thanks! Merged to master.

**Pull request merged**